data/functions/itemized.sql

```diff
@@ -15,3 +121,22 @@ begin
     perform retry_processing_schedule_b_records(:START_YEAR_AGGREGATE);
 end
 $$ language plpgsql;
+
+CREATE OR REPLACE FUNCTION rename_table_cascade(table_name TEXT) RETURNS VOID AS $$
```

Comment:
I totally agree with you on that!  However, we noticed that when we had the "proper" names to begin with and used the same names when creating the new temp tables in prep for an update, the indexes would get dropped/overwritten during updates.  This in turn would cause massive performance issues due to backed up queries and filled connection pools because the API calls on the schedule data would grind to a halt without the indexes there.

We found that if we named the indexes with `_tmp` in them at first when creating the new tables/materialized views during an update and then renaming them preventing this from happening.  If this is still confusing or unclear let's hop on a chat to talk further!